### ISSUE TITLE
won't generate *.dylib on macOS with Xcode

### ISSUE THREAD

**Issue by dingzeyuli** [wjakob/nanogui#267]
I tried to compile nanogui library on my mac. I use cmake to generate an Xcode project. The `libglfw_objects` and `nanogui-obj` can be generated. But the final `nanogui` does not show up.

After some digging, I think this might be the reason. 

> Some native build systems may not like targets that have only object files, so consider adding at least one real source file to any target that references `$<TARGET_OBJECTS:objlib>`.

Sources: [[1](https://gitlab.kitware.com/cmake/cmake/issues/16524
)] [[2](https://cmake.org/cmake/help/v3.7/command/add_library.html?#object-libraries)]

My workaround is to add a blank.c file in the nanogui src and modify the CMakeLists.txt:
```
add_library(nanogui
${NANOGUI_LIBRARY_TYPE}
blank.c
$<TARGET_OBJECTS:nanogui-obj>
$<TARGET_OBJECTS:glfw_objects>
)
```
The blank.c file is empty, just there to activate the Xcode building system. With this modification, nanogui now compiles with Xcode.

**Comment by svenevs** [comment]
Thanks for the detailed report @dingzeyuli, is this the same as #220?  That one kind of disappeared into the fold, but I believe it achieves the same fix.

**Comment by dingzeyuli** [comment]
Yeah, thanks @svenevs. That pull request has not yet been accepted. So the current nanogui code still fails with Xcode+cmake. That one seems to pass the integration test. Is there any plan to accept it? Thank you.

**Comment by wjakob** [comment]
That PR failed the CI tests, so I never seriously looked at it. I'll write some comments on #220.

**Comment by liminchen** [comment]
Hi Liding, Is the problem you encountered the one below?
I tried your fix but it still doesn't work...
<img width="1136" alt="screenshot 2018-01-25 10 53 31" src="https://user-images.githubusercontent.com/6930359/35406606-3900e406-01be-11e8-8626-856dc58ed261.png">

**Comment by svenevs** [comment]
Fixed in #304.
